serverless/serverless
# Support provider > profile and profile per stage

***Implementing Issue:*** #1940

## What did you implement:

This PR has two improvements:

1. Actually load AWS Profiles properly - while there was code to set the profile name, it wasn't being recognised by the AWS SDK.
2. Add support for setting a profile via the `provider > profile` variable in `serverless.yml`.

In addition to this, I have documented how to support different profiles per stage.

This PR supersedes #1945.

## How did you implement it:

The profile is now set by loading the values from the `~/.aws/credentials` file [according to the docs](http://docs.aws.amazon.com/AWSJavaScriptSDK/guide/node-configuring.html) (see the heading "Using Profiles with the SDK").

The AWS provider plugin now attempts to load the profile specified in `provider > profile`, if no environment variable is set.

## How can we verify it:

* Add a profile name to `serverless.yml` in the `provider > profile` property, and run a `sls` command **without** setting an environment variable (e.g. `AWS_PROFILE`). The config-based property will be used.
* Add a `custom > profiles` configuration section and update your `provider > profile` property to use variables (like the example given in the updated docs for [provider account setup](./docs/guide/provider-account-setup.md)). Run an `sls` command specifying different stages and verify the that the correct profile is used.

## Todos:
